api/mongo_helper.py

```diff
@@ -1,71 +1,139 @@
+from logging import Logger
 from bson.objectid import ObjectId
-from datetime import date, datetime
+from datetime import datetime
 import pymongo
 from typing import List
+from pymongo.database import Database
+from pymongo.mongo_client import MongoClient
 
 from api.itask_data_manager import ITaskDataManager
 from api.app_config import AppConfig
 from models.card import Card
 from models.card_list import CardList
 
 class MongoHelper(ITaskDataManager):
-    def __init__(self, app_config: AppConfig):
-        self.config = app_config
-        self.client = pymongo.MongoClient(app_config.MONGO_CONNECTION_STRING)
-        self.db = self.client[app_config.MONGO_DB_NAME]
+    def __init__(self, app_config: AppConfig, logger: Logger):
+        self.logger = logger
+        self.client = self._get_mongo_client(app_config.MONGO_CONNECTION_STRING)
+        self.db = self._get_mongo_db(app_config.MONGO_DB_NAME)
         self.board_metadata = self.db['board-metadata']
         self.cards = self.db['cards']
         self.lists = self.db['lists']
         self.initialise_db()
 
     def get_cards(self) -> List[Card]:
-        mongo_card_array = self.cards.find()
-        return [Card.mongo_dict_to_card(mongo_card) for mongo_card in mongo_card_array]
+        self.logger.debug('Getting cards...')
+        try:
+            mongo_card_array = self.cards.find()
+            return [Card.mongo_dict_to_card(mongo_card) for mongo_card in mongo_card_array]
+        except Exception as e:
+            self.logger.error('Error getting cards. Error message: %s', e)
     
     def get_card(self, id) -> Card:
-        mongo_card = self.cards.find({'_id': ObjectId(id)})
-        return Card.mongo_dict_to_card(mongo_card)
+        self.logger.debug('Getting Card %s ...', id)
+        try:
+            mongo_card = self.cards.find({'_id': ObjectId(id)})
+            return Card.mongo_dict_to_card(mongo_card)
+        except Exception as e:
+            self.logger.error('Error getting Card %s. Error message: %s', id, e)
```

Comment:
Worth noting that although you're logging these errors, you then have this function not return anything which will likely cause a follow on error for whoever called it! Instead we might want throw the exception again, to let it bubble up and be handled elsewhere (where it could e.g. trigger showing an error page), or if appropriate we could return an empty list (but that could be very confusing!)

Often we'll let the exceptions occur here without specific logging, but ensure they're caught at a higher level - as long as a stack trace is provided we can still determine exactly where the error occured. [See e.g. this SO discussion](https://stackoverflow.com/questions/29332056/global-error-handler-for-any-exception) where we could log the exception & return an appropriate error page